hbase-server/src/test/java/org/apache/hadoop/hbase/ccsmap/TestCompactedConcurrentSkipList.java

```diff
@@ -0,0 +1,1842 @@
+          for (AbstractBehavior bh : bhs) {
+            totalw += bh.getWeight();
+            wm.put(totalw, bh);
+            System.out.println("Put " + bh.getName() + " into at " + type
+              + " of weight " + totalw);
+          }
+          totalWeight.put(type, totalw);
+          weightedBehaviors.put(type, wm);
+        }
+      }
+
+      public Behavior nextBehavior(InputType inputType) {
+        int totalw = totalWeight.get(inputType);
+        int l = getRandom().nextInt(totalw);
+        Entry<Integer, AbstractBehavior> entry = weightedBehaviors.get(
+          inputType).higherEntry(l);
+        return entry.getValue();
+      }
+    }
+
+    static public void initData(MapContext mc, int count) {
+      for (int i = 0; i < count; ++i) {
+        KVPair<Integer, String> kv = getRandomKV(mc);
+        mc.mainMap1.put(kv.key(), kv.value());
+        mc.mainMap2.put(kv.key(), kv.value());
+      }
+    }
+  } // end of behavior collctions
+
+  static class Performace {
+    Map<String, Integer> behaviorToOps = new ConcurrentHashMap<>();
+    Map<String, Long> behaviorToLatency = new ConcurrentHashMap<>();
+
+    void addOps(String behavior, long latencyInNs) {
+      Integer o = behaviorToOps.get(behavior);
+      o = o == null ? 1 : o + 1;
+      behaviorToOps.put(behavior, o);
+
+      Long l = behaviorToLatency.get(behavior);
+      l = l == null ? latencyInNs : l + latencyInNs;
+      behaviorToLatency.put(behavior, l);
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      for (Entry<String, Integer> e : behaviorToOps.entrySet()) {
+        int ops = e.getValue();
+        long lat = behaviorToLatency.get(e.getKey()) / ops / 1000;
+        sb.append("Ops:").append(ops).append('\t').append("AvgLatency:")
+          .append(lat).append('\t').append("B:").append(e.getKey())
+          .append('\n');
+      }
+      return sb.toString();
+    }
+  }
+
+  @Test
+  public void testRandomBehavior() {
+    final int RANGE = 32768;
+    final int INIT_DATA = 50000;
+    final int COUNT = 10000000;
+    MapContext mc = new MapContext(createMapForTest(true),
+      createMapForStand(), RANGE);
+    BehaviorCollections.initRandom(RANGE);
+    BehaviorCollections.initData(mc, INIT_DATA);
+    System.out.println("There are " + mc.mainMap2.size()
+      + " element in this map");
+    Performace performance = new Performace();
+    BehaviorCollections.RandomBehavior rb = new BehaviorCollections.RandomBehavior(false);
+    for (int i = 0; i < COUNT; ++i) {
+      Behavior b = rb.nextBehavior(mc.type);
+      if (mc.type != InputType.MAIN_MAP) {
+        mc.viewOps += 1;
+      }
+      long start = System.nanoTime();
+      b.doBehavior(mc);
+      long lat = System.nanoTime() - start;
+      performance.addOps(b.getName(), lat);
+      if (mc.viewOps > 20) { // up to 20 view operations
+        mc.reset();
+      }
+    }
+    System.out.println("There are " + mc.mainMap2.size()
```

Comment:
Is it possible to establish baseline so that we can make some assertion here ?
I am not sure who would periodically examine these output .